web_app_drawer/static/src/js/web_app_drawer.js

```diff
@@ -10,24 +10,24 @@ odoo.define('web_app_drawer.app_drawer', function(require) {
     var core = require('web.core');
     
     Menu.include({
-        
+        // Force all_outside to prevent app icons from going into more menu
         reflow: function() {
             this._super('all_outside');
         },
-        
     });
     
     SearchView.include({
         
-        // ReImplement core toggle_visibility to not focus when mobile
+        // Prevent focus of search field on mobile devices
         toggle_visibility: function (is_visible) {
-            this.do_toggle(!this.headless && is_visible);
-            if (this.$buttons) {
-                this.$buttons.toggle(!this.headless && is_visible && this.visible_filters);
-            }
-            if (!this.headless && is_visible && !this.isMobile()) {
-                this.$('div.oe_searchview_input').last().focus();
-            }
+            var self = this,
+                $searchField = $('div.oe_searchview_input').last();
+            $searchField.one('focus', function(){
```

Comment:
Well, technically JS is designed in a way that almost forces you to make hacks.

Let me explain a bit about this. `this` is hackable in JS. And it is hacked in almost every framework to be something you might not expect. For example, in jQuery events, jQuery takes care of making `this` to be the DOM element where the event is being triggered, and in jQuery deferred/promises, the promise object.

So no matter where you define your method, you could find that it gets triggered in a different *context*. That's why we need this kind of tricks.

There are a big amount of ways of proxying:

- `var self = this`. This is quite common because it's what first comes to mind when you don't know other methods to do it. BTW we use `self` probably because that's the standard in Python, but actually [there's a default `self` in every browser window](http://www.w3schools.com/jsref/prop_win_self.asp). If using this, I prefer `var _this = this`. Anyway `self` works too, so don't worry too much (unless you want to use `window.self` inside your method, of course :wink:).
- [`$.proxy`](http://api.jquery.com/jQuery.proxy/), the jQuery solution to the problem. 1st arg is the method, 2nd is the context. You can use this for events, promises, etc. Quite handy.
- [`_.bind`](http://underscorejs.org/#bind), the underscore solution, similar to jQuery's.
- Standard JS solutions:
  - [`bind`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/bind)
  - [`apply`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/apply)
  - [`call`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/call)
  - [`Proxy`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy)

Sadly in the JS world, it's common better to use framework tools than native ones, since they take care of different implementations and browser compatibility. When we have such tools to do the hackery, using `self=this` seems cumbersome. Actually you are doing the same as `$.proxy`, and most code we write in Odoo is jQuery-based, so I recommend that approach.

So, do you have to refactor everything? Actually not. All solutions work. But getting the habit of using proxies implicitly implants the habit of not declaring lambdas (the lambda is made for you), which leads to more structured, extensible and readable code, and also makes readers think you know the inners of JS like a pro! :laughing: